R/zchunk_LB165.ag_water_R_C_Y_GLU_irr.R

```diff
@@ -0,0 +1,407 @@
+
+    L165.ag_Water_ctry_MHcropX %>%
+      left_join_error_no_match(L165.rfdProd_t_ctry_crop, by = c("iso", "GTAP_crop")) %>%
+      mutate(GreenRfd_m3kg = if_else(rfdProd_t != 0, GreenRfd_thousm3 / rfdProd_t, 0)) ->
+      L165.ag_Water_ctry_MHcropX
+
+    # The grid-based and nation-based tables of green and blue water coefs and volumes for irrigated
+    # and rainfed crops can now be combined to provide estimates of green and blue water coefs for all
+    # available countries, crops, and GLUs (original lines 212-229)
+    # First we need to write out the MH2014 coefficients to all applicable GLUs
+    L165.ag_Prod_t_ctry_crop_GLU %>%
+      select(iso, GTAP_crop, GLU) %>%
+      filter(iso %in% L165.Mekonnen_Hoekstra_Rep47_A2$iso,
+             GTAP_crop %in% L165.ag_Water_ctry_MHcropX$GTAP_crop) %>%
+      left_join_error_no_match(select(L151.ag_irrProd_t_ctry_crop, iso, GTAP_crop, GLU, irrProd),
+                               by = c("iso", "GTAP_crop", "GLU")) %>%
+      rename(irrProd_t = irrProd) %>%
+      left_join_error_no_match(select(L151.ag_rfdProd_t_ctry_crop, iso, GTAP_crop, GLU, rfdProd),
+                               by = c("iso", "GTAP_crop", "GLU")) %>%
+      rename(rfdProd_t = rfdProd) %>%
+      left_join_error_no_match(select(L165.ag_Water_ctry_MHcropX, iso, GTAP_crop, BlueIrr_m3kg, GreenIrr_m3kg, GreenRfd_m3kg),
+                               by = c("iso", "GTAP_crop")) ->
+      L165.ag_Water_ctry_MHcropX_GLU
+
+    # Combine with the grid-based estimates (original lines 231-241)
+    L165.ag_Water_ctry_MHcropX_GLU %>%
+      bind_rows(L165.ag_Water_ctry_MHcrop_GLU) %>%
+      select(iso, GTAP_crop, GLU, irrProd_t, rfdProd_t, BlueIrr_m3kg, GreenIrr_m3kg, GreenRfd_m3kg) %>%
+      # calculate green and blue water quantities by all available countries, crops, GLUs, and irrigated/rainfed
+      mutate(BlueIrr_thousm3 = BlueIrr_m3kg * irrProd_t,
+             GreenIrr_thousm3 = GreenIrr_m3kg * irrProd_t,
+             GreenRfd_thousm3 = GreenRfd_m3kg * rfdProd_t) %>%
+      # match in GCAM regions and commodities for aggregation
+      left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+      left_join_keep_first_only(select(FAO_ag_items_PRODSTAT, GTAP_crop, GCAM_commodity), by = "GTAP_crop") ->
+      L165.ag_Water_ctry_crop_GLU
+
+    # Original lines 243-265
+    # NOTE: Fodder crops are not in the M+H inventories. They instead have been assigned a
+    # corresponding crop that is in the M+H inventories and at this point are multiplied by
+    # an exogenous fodder:crop adjustment factor. This factor is designed to reflect that
+    # fodder crop masses include the whole plant, often harvested green (~70% water). As such,
+    # their water coefficients will tend to be a good deal lower than the corresponding crop
+    # to which they are assigned. The specific number is based on analysis of a virtual water
+    # content dataset (Chapagain 2004) that included fodder crops.
```

Comment:
not really a review comment so much as a thought: does it make sense for us to essentially write a reference list for the data system? For example, Chapagain 2004 and the context is probably enough to find that source, but maybe we should compile everything in one place.